File: temp.py
def calcular_media_temperaturas():
  """
  Calcula a soma e a média de uma série de temperaturas fornecidas pelo usuário.
  O usuário decide quantas temperaturas serão inseridas.
  """
  temperaturas = []
  
  while True:
    try:
      quantidade = int(input("Quantas temperaturas você quer inserir para calcular a média? "))
      if quantidade > 0:
        break
      else:
        print("Por favor, digite um número maior que zero.")
    except ValueError:
      print("Entrada inválida. Por favor, digite um número inteiro.")

  for i in range(quantidade):
    while True:
      try:
        temp = float(input(f"Digite a temperatura {i + 1}: "))
        temperaturas.append(temp)
        break
      except ValueError:
        print("Entrada inválida. Por favor, digite um número.")

  if temperaturas:
    soma_temperaturas = sum(temperaturas)
    media_temperaturas = soma_temperaturas / len(temperaturas)

    print(f"\nA soma total das temperaturas é: {soma_temperaturas:.2f}°C")
    print(f"A média das temperaturas é: {media_temperaturas:.2f}°C")
  else:
    print("Nenhuma temperatura foi inserida para o cálculo.")

# Executar o programa
if __name__ == "__main__":
  calcular_media_temperaturas()

File: temp.c
#include <stdio.h>

int main() {
    int quantidade, i;
    float temperatura, soma_temperaturas = 0.0, media_temperaturas;

    // Pede ao usuário a quantidade de temperaturas a serem inseridas
    printf("Quantas temperaturas voce quer inserir para calcular a media? ");
    scanf("%d", &quantidade);

    // Valida se a quantidade é positiva
    if (quantidade <= 0) {
        printf("Por favor, digite um numero maior que zero.\n");
        return 1; // Retorna um código de erro
    }

    // Loop para ler as temperaturas e somá-las
    for (i = 1; i <= quantidade; i++) {
        printf("Digite a temperatura %d: ", i);
        scanf("%f", &temperatura);
        soma_temperaturas += temperatura;
    }

    // Calcula a media
    media_temperaturas = soma_temperaturas / quantidade;

    // Exibe os resultados
    printf("\nA soma total das temperaturas e: %.2f C\n", soma_temperaturas);
    printf("A media das temperaturas e: %.2f C\n", media_temperaturas);

    return 0; // Retorna 0 para indicar sucesso
}